Adapter.C

```diff
@@ -923,6 +924,60 @@ void preciceAdapter::Adapter::setupCheckpointing()
     ACCUMULATE_TIMER(timeInCheckpointingSetup_);
 }
 
+void preciceAdapter::Adapter::pruneCheckpointedFields()
+{
+    // Check if checkpointed fields exist in OpenFOAM registry
+    // If not, remove them from the checkpointed fields vector
+
+    word fieldName;
+    std::vector<word> regFields;
+    std::vector<word> toRemove;
+
+#undef doLocalCode
+#define doLocalCode(GeomField, GeomField_, GeomFieldCopies_)                                                           \
```

Comment:
Add a comment on what the three arguments are.
Or rename `GeomField` to `GeomFieldType`?